### DISCUSSION TITLE
G7 low battery warning

### DISCUSSION THREAD

**Discussion by batachman** [NightscoutFoundation/xDrip#3408] [Q&A]
I started using xDrip recently and it has been so much better than the Dexcom app. One thing that came up today though is I received a warning about transmitter battery low, but my sensor is only 5 days old. I've tried to find something documented showing how low it actually is, but can't find anything. I doubt there should be any reason a sensor would run out of battery sooner in xDrip, but would appreciate some info if anyone knows if this is a false warning and I should be good for the rest of the sensor life, or at least at what voltage it fails. Wouldn't think I would need any type of warning on a device that only lasts 10 days unless there really was an issue. 
![Screenshot_20240328_160126_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/36554627/cc40e29e-20df-4fbc-a532-1e8cbf9500ff)

**Comment by Navid200** [comment]
Those voltages are extremely low.
Hopefully, it will run 5 more days.

We have made a lot of updates for G7 that you can benefit from if you update to the latest Nightly.

**Comment by Navid200** [comment]
Assuming you have been using xDrip with this for the past 5 days, you can see how the voltage has been changing over the past 5 days by doing the following:

Go to Logs: https://navid200.github.io/xDrip/docs/Logs.html
Enable all categories.
Tap on the magnifying glass symbol to search and search for the following.

volt

You can then see Voltage B listed for the past 10 days.  Of course, you are only interested for those values for the past 5 days.  What are they?
If you put them in an Excel sheet and create a graph to show voltage B over time, you can see how fast Voltage B is dropping.

If you buy a device and let it sit on a shelf for a very long time, the battery will gradually leak and eventually die.

**Reply by batachman** [reply]
Oddly, other than the first day, it isn't showing much of a drop so hopefully it will be fine. 
<img width="912" alt="Screenshot 2024-03-29 at 6 26 50 AM" src="https://github.com/NightscoutFoundation/xDrip/assets/36554627/c2fffeb6-6deb-43d8-8bb0-70def3eded3a">

**Reply by Navid200** [reply]
There are two dates printed on a G7 box.  The first is the manufacture date.  The second is the best before date.
What re those on the box?

**Reply by batachman** [reply]
2023-09-01
2025-02-28

**Reply by Navid200** [reply]
OK,

It looks like we may have been wrong about the reported values.  
Thanks for reporting this.  It should be fine.

**Reply by Navid200** [reply]
Did it work the remaining 5 days?

What do you see for battery voltages of your next sensor?

**Reply by batachman** [reply]
It lasted the remaining days. Not a lot of data on the second sensor yet, but the values are slightly higher. Did take a bit for the first one to drop though. 

<img width="1431" alt="Screenshot 2024-04-03 at 3 15 34 PM" src="https://github.com/NightscoutFoundation/xDrip/assets/36554627/4009c4cf-6a35-49bd-acfc-3783b19f7ed8">

**Reply by Navid200** [reply]
Would you mind telling me where the device was purchased from?
If you don't want to, that's fine.
If you prefer to provide it confidentially, please send an email to xDrip dot Navid at gmail dot com

Thanks

**Reply by batachman** [reply]
The first sensor was from a Costco pharmacy in the US. The second was sent from Dexcom as a replacement for one that failed.

**Reply by Navid200** [reply]
Thanks

**Reply by Navid200** [reply]
I am in Canada. 
I have had 16 devices so far.  None with a voltage B less than 260 over the first 10 days of life.

**Reply by Navid200** [reply]
I have been monitoring the voltages of all of these 16 devices even after 10 days.
Voltage B for none of them dropped below 250 up to day 60.

Those that have reached day 90, voltage B is still higher than 230.
I will continue to monitor hoping to have a better understanding of how the voltages behave.

I have done this to make sure we have correct warning levels.

This is why I was so surprised when I saw your readings.  It is an anomaly.

**Comment by MisterMcDuck** [comment]
FWIW I started a sensor recently (which didn't seem to work at first) and it reported
Voltage A: 292
Voltage B: 214

and is currently reporting successfully for 2 hours now. It had a few months before the expiration date was reached.

**Reply by Navid200** [reply]
We lowered the threshold significantly not long ago: https://navid200.github.io/xDrip/docs/ReleaseNotes.html 

We may need to further lower it.  But, we need to give it some time to see how many of these come up.
